src/association/association_internal.rs

```diff
@@ -712,13 +714,28 @@ impl AssociationInternal {
             ..Default::default()
         };
 
+        let unrecognized_params_from_init = i.params.iter().filter_map(|param| {
+            if let ParamType::Unknown { param_type } = param.header().typ {
+                let needs_to_be_reported = ((param_type >> 14) & 0x01) == 1;
+                if needs_to_be_reported {
+                    let wrapped: Box<dyn Param + Send + Sync> = Box::new(ParamUnrecognized::wrap(param.clone()));
+                    Some(wrapped)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        }).collect();
+
```

Comment:
This `ParamUnrecognized` throws me. The name seems semantically identical to `ParamUnknown`, but there must be some reason we need both. Can that reason/distinction be brought out in at least doc comments?

Reply:
Oh yeah I can definitely add some comments on that. 

We need both because at the parsing stage we need a way to represent params that we just don't know the type of. That is `ParamUnknown`. It transparently wraps any valid param. I you marshal it again it turns into the same bytes as the one you just unmarshaled.

The `ParamUnrecognized` is an actual separate Param type, with it's own unique type id and everything. It is used to report unrecognized params to the sender. If you marshal that it wraps the contained Param (probably a `ParamUnknown`) in a separate param header marking it as unrecognized.

Reply:
makes sense. let stick that explanation in the code. 

thanks!

Reply:
Done :)